src/browser/fake_network.js

```diff
@@ -43,6 +58,220 @@ function iptolong(parts) {
     return parts[0] << 24 | parts[1] << 16 | parts[2] << 8 | parts[3];
 }
 
+class Uint8Stream
+{
+    /**
+     * @param {number} initial_capacity
+     * @param {number} maximum_capacity
+     */
+    constructor(initial_capacity, maximum_capacity)
+    {
+        this.maximum_capacity = maximum_capacity;
+        this.tail = 0;
+        this.head = 0;
+        this.length = 0;
+        this.buffer = new Uint8Array(initial_capacity);
+    }
+
+    /**
+     * @param {number} capacity
+     */
+    resize(capacity)
+    {
+        const new_buffer = new Uint8Array(capacity);
+        this.peek(new_buffer, this.length);
+        this.tail = 0;
+        this.head = this.length;
+        this.buffer = new_buffer;
+    }
+
+    /**
+     * @param {Uint8Array} src_array
+     */
+    write(src_array)
+    {
+        const src_length = src_array.length;
+        const total_length = this.length + src_length;
+        let capacity = this.buffer.length;
+        if(capacity < total_length) {
+            while(capacity < total_length) {
+                capacity *= 2;
```

Comment:
This can loop forever if capacity is zero. I'd suggest adding `Math.min(capacity, 16)` or `console.assert(capacity > 0)`

Reply:
I changed it to guarantee a minimum capacity of 16 bytes.